.github/workflows/ci.yml

```diff
@@ -0,0 +1,102 @@
+name: Docker Image CI
+
+on:
+    push:
+        branches:
+            - main
+    pull_request:
+        branches:
+            - main
+
+jobs:
+    mindmapper-backend-build:
+        runs-on: ubuntu-latest
+
+        steps:
+            - uses: actions/checkout@v2
+
+            - name: Set up QEMU
+              uses: docker/setup-qemu-action@v1
+
+            - name: Set up Docker Buildx
+              uses: docker/setup-buildx-action@v1
+
+            - name: Login to GitHub Container Registry
+              uses: docker/login-action@v1
+              with:
+                  registry: ghcr.io
+                  username: ${{ github.repository_owner }}
+                  password: ${{ secrets.GITHUB_TOKEN }}
+
+            - name: Build and export to Docker
+              uses: docker/build-push-action@v2
+              with:
+                  context: .
+                  target: development
+                  push: true
+                  tags: |
+                      ghcr.io/b310-digital/mindmapper/ci:${{ github.sha }}-dev
+            #   outputs: type=docker,dest=/tmp/myimage.tar
+
+            # - name: Upload artifact
+            #   uses: actions/upload-artifact@v2
+            #   with:
+            #       name: myimage
+            #       path: /tmp/myimage.tar
+
+    mindmapper-backend-lint-2:
```

Comment:
why the name lint-2?

Reply:
ah ok so there are two lint jobs, one based on docker, one not? do you want to keep both or what's the idea here?

Reply:
This branch is not ready yet.

The idea is to build the docker image during the ci co build (<= this works already). And then use this docker image to execute the other jobs, e.g. `lint` and 1test:e2e` <= I have to work on this a little bit further.

For the time being I am keeping the existing jobs. When I am done with the configuration of the ci cd pipeline. I will remove the others.